addons/binding/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/handler/NikoHomeControlBridgeHandler.java

```diff
@@ -0,0 +1,266 @@
+
+        logger.debug("Niko Home Control: bridge handler port {}", port);
+
+        if (config.get(CONFIG_HOST_NAME) != null) {
+            try {
+                // If hostname or address was provided in the configuration, try to use this to for bridge and give
+                // error
+                // when hostname parameter was not valid.
+                // No hostname provided is a valid configuration, therefore allow null addr to pass through.
+                logger.debug("Niko Home Control: configure bridge with host parameter {}",
+                        config.get(CONFIG_HOST_NAME));
+                addr = InetAddress.getByName((String) config.get(CONFIG_HOST_NAME));
+            } catch (UnknownHostException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                        "Niko Home Control: cannot resolve bridge IP with hostname " + config.get(CONFIG_HOST_NAME));
+                return;
+            }
+        } else if (config.get(CONFIG_BROADCAST_ADDRESS) != null) {
+            try {
+                logger.debug("Niko Home Control: configure bridge with broadcast parameter {}",
+                        config.get(CONFIG_BROADCAST_ADDRESS));
+                broadcastAddr = InetAddress.getByName((String) config.get(CONFIG_BROADCAST_ADDRESS));
+            } catch (UnknownHostException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                        "Niko Home Control: cannot resolve broadcast address " + config.get(CONFIG_BROADCAST_ADDRESS));
+                return;
+
+            }
+        } else {
+            logger.debug("Niko Home Control: try to auto-discover bridge address");
+        }
+
+        createCommunicationObject(addr, port, broadcastAddr);
+
+    }
+
+    /**
+     * Create communication object to Niko Home Control IP-interface and start communication.
+     * Trigger discovery when communication setup is successful.
+     *
+     * @param addr IP address of Niko Home Control IP-interface or null
+     * @param port
+     */
+    private void createCommunicationObject(InetAddress addr, int port, InetAddress broadcastAddr) {
+
+        Configuration config = this.getConfig();
+
+        scheduler.submit(new Runnable() {
+
+            @Override
+            public void run() {
+                try {
+                    nhcComm = new NikoHomeControlCommunication(addr, broadcastAddr);
+
+                    nhcComm.startCommunication(port);
+                    if (!nhcComm.communicationActive()) {
+                        throw new IOException("Niko Home Control: communication socket error");
+                    }
+
+                    // Set callback from NikoHomeControlCommunication object to this bridge to be able to take bridge
+                    // offline when non-resolvable communication error occurs.
+                    setBridgeCallBack();
```

Comment:
I think this makes it more complex. The method can access this and pass it on to another object. As this line is in a job, I don't have straight access to the object anymore. It would become something like:
`nhcComm.setBridgeCallBack((NikoHomeControlBridgeHandler) thing.getHandler())`
I am not convinced this is easier, unless there is a simpler way to write this.